classes/Environment.php

```diff
@@ -39,13 +44,25 @@ public static function testing(): self
         return new self(self::TYPE_TESTING);
     }
 
+    /**
+     * @throws \InvalidArgumentException
+     *
+     * @return self
+     */
     public static function fromEnvironmentVariable(): self
```

Comment:
Maybe it makes sense to add an argument `$server` here?

Then throwing an `InvalidArgumentException` makes sense, and we don't need to access globals from within the method.